### ISSUE TITLE
ConditionalDimension: Condition indexing not shifted by offset.

### ISSUE THREAD

**Issue by georgebisbas** [devitocodes/devito#1273]
The condition generated is not shifted by the halo offset. (No lowering is happening, neither indexification.)

```
from devito import Grid, ConditionalDimension, Function, Eq, Operator
from devito.symbolics import CondEq
import pdb

grid = Grid(shape=(4, 4))  # Define grid

# Define Function
g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
x, y = grid.dimensions

ci = ConditionalDimension(name='ci', parent=y, condition=CondEq(g[x, y], 0))

f = Function(name='f', shape=grid.shape, dimensions=(x, ci))

op = Operator(Eq(f[x, ci], g[x, y] + 1))
print(op.ccode)
op.apply()
pdb.set_trace()
```

generates:

```
#pragma omp parallel num_threads(nthreads_nonaffine)
  {.
.
.
  for (int y = y_m; y <= y_M; y += 1)
      {
        if (g[x][y] == 0)
        {
          f[x + 1][y + 1] = g[x + 1][y + 1] + 1;
        }
      }
    }
  }
.
.

```

while one would expect `if (g[x+1][y+1] == 0)`

**Comment by georgebisbas** [comment]
From @FabioLuporini : https://devitocodes.slack.com/archives/C3W8GF3H9/p1588692793038800
```
@George Bisbas hook for 1273: https://github.com/devitocodes/devito/blob/master/devito/operator/operator.py#L300
hints. There are two issues:
- (AFAIR) not searching for ConditionalDimensions and their condition
- even if we did, our objects are immutable so we can’t really change the condition  directly
```

**Comment by FabioLuporini** [comment]
fixed by #1276
